iceberg-catalog-migrator/gradle/libs.versions.toml

```
@@ -62,8 +62,15 @@ google-java-format = { module = "com.google.googlejavaformat:google-java-format"
 guava = { module = "com.google.guava:guava", version.ref = "guava" }
 hadoop-aws = { module = "org.apache.hadoop:hadoop-aws", version.ref = "hadoop" }
 hadoop-common = { module = "org.apache.hadoop:hadoop-common", version.ref = "hadoop" }
+iceberg-api = { module = "org.apache.iceberg:iceberg-api", version.ref = "iceberg" }
+iceberg-aws = { module = "org.apache.iceberg:iceberg-aws", version.ref = "iceberg" }
+iceberg-azure = { module = "org.apache.iceberg:iceberg-azure", version.ref = "iceberg" }
+iceberg-common = { module = "org.apache.iceberg:iceberg-common", version.ref = "iceberg" }
+iceberg-core = { module = "org.apache.iceberg:iceberg-core", version.ref = "iceberg" }
 iceberg-dell = { module = "org.apache.iceberg:iceberg-dell", version.ref = "iceberg" }
-iceberg-spark-runtime = { module = "org.apache.iceberg:iceberg-spark-runtime-3.5_2.12", version.ref = "iceberg" }
+iceberg-gcp = { module = "org.apache.iceberg:iceberg-gcp", version.ref = "iceberg" }
+iceberg-hive-metastore = { module = "org.apache.iceberg:iceberg-hive-metastore", version.ref = "iceberg" }
+iceberg-nessie = { module = "org.apache.iceberg:iceberg-nessie", version.ref = "iceberg" }
```

```suggestion
iceberg-bom = { module = "org.apache.iceberg:iceberg-bom", version.ref = "iceberg" }
```
It's less work to use the bom-approach and refer to individual deps in [this way](https://github.com/apache/polaris/blob/4a80c513d142c6f9211bfc1e431794a7dd3ac1e9/polaris-core/build.gradle.kts#L28-L30).

---

BTW: I suspect it's easier to split the dependency changes out into a separate PR, as I suspect the license/notice work requires a few more iterations.